ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java

```diff
@@ -544,37 +534,18 @@ public void run() {
       }
     }
 
-    private boolean isOptimizerTimeout(String token) {
-      return Optional.ofNullable(authOptimizers.get(token))
-          .map(
-              optimizer ->
-                  optimizer.getTouchTime() + optimizerTouchTimeout < System.currentTimeMillis())
-          .orElse(true);
-    }
-
-    private void dealSuspendingTasks(String token, boolean isOptimzerExpired) {
-      OptimizingQueue queue = getQueueByToken(token);
-      queue
-          .collectTasks(buildSuspendingPredication(token, isOptimzerExpired))
-          .forEach(
-              task -> {
-                LOG.info(
-                    "Task {} is suspending, since it's optimizer is expired, put it to retry queue, optimizer {}",
-                    task.getTaskId(),
-                    task.getResourceDesc());
-                // optimizing task of suspending optimizer would not be counted for retrying
-                queue.retryTask(task, false);
-              });
+    private void retryTask(TaskRuntime task, OptimizingQueue queue) {
+      LOG.info(
+          "Task {} is suspending, since it's optimizer is expired, put it to retry queue, optimizer {}",
+          task.getTaskId(),
+          task.getResourceDesc());
+      // optimizing task of suspending optimizer would not be counted for retrying
+      queue.retryTask(task, false);
     }
 
-    private void dealOptimizer(String token, boolean isOptimzerExpired) {
-      if (isOptimzerExpired) {
-        LOG.info("Optimizer {} has been suspended, unregister it", token);
-        unregisterOptimizer(token);
-      } else {
-        OptimizerInstance optimizer = authOptimizers.get(token);
-        detectTimeout(optimizer);
-      }
+    private boolean checkOptimizerTimeout(OptimizerInstance optimizer) {
+      return Objects.equals(optimizer, authOptimizers.get(optimizer.getToken()))
+          && optimizer.getTouchTime() + optimizerTouchTimeout > System.currentTimeMillis();
```

Comment:
I think it should be less than here. If the last touch time plus the timeout interval is less than the current time, it means that there hasn't been another touch after exceeding the timeout interval.
`optimizer.getTouchTime() + optimizerTouchTimeout  <   System.currentTimeMillis();`

Reply:
The logic has been tested in TestDefaultOptimizingQueue, and as you mentioned, this is fixed.

Check it out plz